From 30455d58d268e2dfb02274ac6b2b9ee05880b7d9 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Mon, 12 Oct 2026 02:04:57 +0200
Subject: [PATCH] avcodec/aac: do not keep the frame pointer after decoding

Fixes: use after free
Fixes: MbYbttK2ue35
Regression since: eee5fa0808
Found during triage of the review of https://code.ffmpeg.org/FFmpeg/FFmpeg/pulls/24678
Use after free Replicated through UnModified FFmpeg with ASAN

(cherry picked from commit 30455d58d268e2dfb02274ac6b2b9ee05880b7d9)
---
 libavcodec/aac/aacdec.c      | 1 +
 libavcodec/aac/aacdec_latm.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/libavcodec/aac/aacdec.c b/libavcodec/aac/aacdec.c
index 66ab648016d..700f16c56a9 100644
--- a/libavcodec/aac/aacdec.c
+++ b/libavcodec/aac/aacdec.c
@@ -2612,6 +2612,7 @@ static int aac_decode_frame(AVCodecContext *avctx, AVFrame *frame,
     default:
         err = aac_decode_frame_int(avctx, frame, got_frame_ptr, &gb, avpkt);
     }
+    ac->frame = NULL;
     if (err < 0)
         return err;
 
diff --git a/libavcodec/aac/aacdec_latm.h b/libavcodec/aac/aacdec_latm.h
index 398d40741bd..d9e836e949e 100644
--- a/libavcodec/aac/aacdec_latm.h
+++ b/libavcodec/aac/aacdec_latm.h
@@ -308,6 +308,7 @@ static int latm_decode_frame(AVCodecContext *avctx, AVFrame *out,
     default:
         err = aac_decode_frame_int(avctx, out, got_frame_ptr, &gb, avpkt);
     }
+    latmctx->aac_ctx.frame = NULL;
     if (err < 0)
         return err;
 
-- 
2.53.0

